util/roomValidate.js

```diff
@@ -0,0 +1,43 @@
+module.exports = function (rooms, config) {
+    return {
+        username: function (username, room, callback) {
+            var error;
+            if (typeof username === 'undefined'
+                || username === ''
+                || username === null
+                || username.match(/server/i)) {
+                error = {
+                    name: 'adduserfail',
+                    arg: 'Invalid username: ' + username
+                };
+                return callback(error, false);
```

Comment:
We should check the validity of `callback` before calling it: `if (typeof callback === 'function') { callback(...); }`